src/command_line_parser.cc

```diff
@@ -1791,16 +1849,32 @@ CLParser::ParseCommandLine(int argc, char** argv)
   }
 
   // Overriding the max_threads default for request_rate search
-  if (!params_->max_threads_specified && params_->targeting_concurrency()) {
+  if (!params_->max_threads_specified &&
+      params_->inference_load_mode == InferenceLoadMode::Concurrency) {
     params_->max_threads =
         std::max(DEFAULT_MAX_THREADS, params_->concurrency_range.end);
   }
 
-  if (params_->using_custom_intervals) {
+  if (params_->inference_load_mode == InferenceLoadMode::CustomIntervals) {
     // Will be using user-provided time intervals, hence no control variable.
     params_->search_mode = SearchMode::NONE;
   }
 
+  // In fixed schedule mode, the request count is based on the number of
+  // payloads in the user-provided dataset
+  if (params_->inference_load_mode == pa::InferenceLoadMode::FixedSchedule) {
+    for (const auto& input_data_path : params_->user_data) {
+      if (!std::filesystem::is_regular_file(input_data_path)) {
+        throw std::runtime_error(
+            "'" + input_data_path +
+            "' is not a file. When using `--fixed-schedule`, all paths "
+            "passed to --input-data must be files.");
+      }
+    }
+
+    params_->request_count = pa::DataLoader::GetDatasetSize(params_->user_data);
+  }
+
```

Comment:
Previously, request count was set to the size of `--schedule` list, which was known at command line argument parsing time (here).

Now, it is based on the number of payloads in the user-provided input data JSON.

Ideally, this would be parsed closer to where the input data is used, but because of the current design of PA, it was important to know the request count earlier than that point, thus this unideal code to scan the input data to determine the request count.

This is not the ideal solution but making the proper solution would have extended the timeline much farther than I or the customers would have wanted.